src/ring_theory/ideal/operations.lean

```diff
@@ -171,18 +157,11 @@ variables (S : set R) (T : set M)
 
 theorem span_smul_span : (ideal.span S) • (span R T) =
   span R (⋃ (s ∈ S) (t ∈ T), {s • t}) :=
-le_antisymm (smul_le.2 $ λ r hrS n hnT, span_induction hrS
-  (λ r hrS, span_induction hnT
-    (λ n hnT, subset_span $ set.mem_bUnion hrS $
-      set.mem_bUnion hnT $ set.mem_singleton _)
-    ((smul_zero r : r • 0 = (0:M)).symm ▸ submodule.zero_mem _)
-    (λ x y, (smul_add r x y).symm ▸ submodule.add_mem _)
-    (λ c m, by rw [smul_smul, mul_comm, mul_smul]; exact submodule.smul_mem _ _))
-  ((zero_smul R n).symm ▸ submodule.zero_mem _)
-  (λ r s, (add_smul r s n).symm ▸ submodule.add_mem _)
-  (λ c r, by rw [smul_eq_mul, mul_smul]; exact submodule.smul_mem _ _)) $
-span_le.2 $ set.Union₂_subset $ λ r hrS, set.Union₂_subset $ λ n hnT, set.singleton_subset_iff.2 $
-smul_mem_smul (subset_span hrS) (subset_span hnT)
+(map₂_span_span _ _ _ _).trans $ congr_arg _ $ begin
+  simp_rw [linear_map.lsmul_apply],
+  conv_lhs { rw [←set.bUnion_of_singleton S, ←set.bUnion_of_singleton T] },
+  simp_rw [set.image2_Union₂_left, set.image2_Union₂_right, set.image2_singleton],
```

Comment:
The term mode proof is pretty cursed, because `supr` (`⋃`) is defined in terms of `Sup` ...
```suggestion
  ext, split,
  { rintro ⟨s,t,hs,ht,h⟩, exact ⟨_,⟨s,rfl⟩,_,⟨hs,rfl⟩,_,⟨t,rfl⟩,_,⟨ht,rfl⟩,h.symm⟩ },
  { rintro ⟨_,⟨s,rfl⟩,_,⟨hs,rfl⟩,_,⟨t,rfl⟩,_,⟨ht,rfl⟩,h⟩, exact ⟨s,t,hs,ht,h.symm⟩ },
```